lib/crypto/crypto_compat.c

```diff
@@ -158,7 +158,7 @@ crypto_compat_RSA_generate_key()
 {
 	RSA * key;
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x00908000L
```

Comment:
Not certain when you want this to kick in.  In 1.0.2 (and even 0.9.8), `RSA_generate_key()` was inside an `# ifndef OPENSSL_NO_DEPRECATED`:
https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/rsa/rsa.h#L326
https://github.com/openssl/openssl/blob/OpenSSL_0_9_8-stable/crypto/rsa/rsa.h#L276

but it only started giving compiler warnings / errors in OpenSSL 1.1.0